packages/core/src/App/Containers/NotificationsDialog/notifications-list.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Button, Icon, Text } from '@deriv/components';
+import { BinaryLink } from 'App/Components/Routes';
+import { isEmptyObject, toTitleCase } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+
+type TActionProps = ReturnType<typeof useStore>['notifications']['notifications'][0]['action'];
+type TNotificationMessage = ReturnType<typeof useStore>['notifications']['notifications'][0];
+
+const NotificationsList = observer(() => {
+    const { notifications } = useStore();
+    const { notifications: notifications_array, toggleNotificationsModal } = notifications;
+
+    const getNotificationItemIcon = (item: TNotificationMessage) => {
+        const { type } = item;
+        if (['contract_sold', 'info', 'news', 'promotions'].includes(type)) {
+            return 'IcAlertInfo';
+        } else if (type === 'p2p_completed_order') {
+            return 'IcAlertAnnounce';
+        }
+        return `IcAlert${toTitleCase(type)}`;
+    };
+
+    const getButtonSettings = (item: TNotificationMessage): TActionProps | undefined => {
+        const object_key = ['action', 'secondary_btn', 'cta_btn', 'primary_btn'].find(
+            obj_key => !isEmptyObject(item[obj_key as keyof TNotificationMessage])
+        );
+        switch (object_key) {
+            case 'primary_btn':
+                return item.primary_btn;
+            case 'cta_btn':
+                return item.cta_btn;
+            case 'secondary_btn':
+                return item.secondary_btn;
+            case 'action':
+                return item.action;
+            default:
+                return undefined;
+        }
+    };
+
+    return (
+        <React.Fragment>
+            {notifications_array.map(item => (
+                <div className='notifications-item' key={item.key}>
+                    <Text
+                        as='h2'
+                        className='notifications-item__title'
+                        weight='bold'
+                        size='xs'
+                        line_height='m'
+                        color='prominent'
+                    >
+                        {item.type && (
+                            <Icon
+                                icon={getNotificationItemIcon(item)}
+                                className={classNames('notifications-item__title-icon', {
+                                    [`notifications-item__title-icon--${item.type}`]: item.type,
+                                })}
+                            />
+                        )}
+                        {item.header}
+                    </Text>
+                    <div className='notifications-item__message'>{item.message}</div>
+                    <div className='notifications-item__action'>
+                        {!!getButtonSettings(item) && (
+                            <React.Fragment>
+                                {getButtonSettings(item)?.route ? (
```

Comment:
clarification: do we need ? in {getButtonSettings(item)?.route since this entire codeblock will be executed when {!!getButtonSettings(item) is true

Reply:
typescript throws warning anyway. maybe for the code editor to display it clean, we can keep it.
![Screenshot 2023-08-31 at 12 20 40 PM](https://github.com/binary-com/deriv-app/assets/122449658/722d58d7-680e-4f1f-943b-c0d0f9075f02)